.github/workflows/bench.yml

```diff
@@ -35,6 +34,68 @@ jobs:
+            run: |
+              # Execute benchmarking script
+              ./run_benchmarks.sh || { echo "Benchmark generation failed"; exit 1; }
+
+              # Extract benchmark information and update README
+              for file in result1_graphql_*.txt; do
+              if [[ -f $file ]]; then
+                SERVER=$(awk -F ']' '{print $1}' <<< $(awk -F '[' '{print $2}' <<< $(basename "$file" .txt)))
+                REQ_SEC=$(grep -oP '[0-9.]+\s*\|\s*[0-9.]+' "$file" | awk -F '|' '{print $2}' | tr -d '[:space:]')
+                LATENCY=$(grep -oP '[0-9.]+\s*\|\s*[0-9.]+' "$file" | awk -F '|' '{print $3}' | tr -d '[:space:]')
+                
+                sed -i "s/\[$SERVER\] \| [0-9.]\+ \| [0-9.]\+/\[$SERVER\] \| $REQ_SEC \| $LATENCY/g" README.md
+              fi
+            done
+
+              # Commit changes to README
+              git add README.md
+              git commit -m "Update benchmark results" || { echo "Failed to commit changes"; exit 1; }
+              git push || { echo "Failed to push changes"; exit 1; }
+
+              # Extract benchmark information and update README
+              for file in result1_graphql_*.txt; do
+               if [[ -f $file ]]; then
+                 SERVER=$(awk -F ']' '{print $1}' <<< $(awk -F '[' '{print $2}' <<< $(basename "$file" .txt)))
+                 REQ_SEC=$(grep -oP '[0-9.]+\s*\|\s*[0-9.]+' "$file" | awk -F '|' '{print $2}' | tr -d '[:space:]')
+                 LATENCY=$(grep -oP '[0-9.]+\s*\|\s*[0-9.]+' "$file" | awk -F '|' '{print $3}' | tr -d '[:space:]')
+                 
+                 sed -i "s/\[$SERVER\] \| [0-9.]\+ \| [0-9.]\+/\[$SERVER\] \| $REQ_SEC \| $LATENCY/g" README.md
+               fi
+              done
+
+              # Commit changes to README
+              git add README.md
+              git commit -m "Update benchmark results" || { echo "Failed to commit changes"; exit 1; }
+              git push || { echo "Failed to push changes"; exit 1; }
+
+              # Check for benchmark generation failure or README update failure
+              if [ $? -ne 0 ]; then
+                echo "CI failed due to benchmark generation or README update failure"
+                exit 1
+              else
+                echo "CI successful"
+              fi
+
+              - name: Run Benchmarks (PR)
+              if: github.event_name == 'pull_request'
+              run: |
+                # Execute benchmarking script for pull requests
+                ./run_benchmarks.sh || { echo "Benchmark generation failed"; exit 1; }
+
+              - name: Final Checks and Reporting
+              run: |
+                # Check for benchmark generation failure or README update failure
+                if [ $? -ne 0 ]; then
+                  echo "CI failed due to benchmark generation or README update failure"
+                  exit 1
+                else
+                  echo "CI successful"
+                fi
```

Comment:
We're duplicating the benchmark running as we already do it on [these lines](https://github.com/tailcallhq/graphql-benchmarks/pull/43/files#diff-a5e740be96415373789689f814583e93ff2a8f05eae6481e94505fd6cb6bc6a7L29-L36).

The error that we have is regarding permissions rather than running the bench or pushing it, as you can see [here](https://github.com/tailcallhq/graphql-benchmarks/actions/runs/6942190533/job/18884599954)
```
remote: error: GH006: Protected branch update failed for refs/heads/main.        
remote: error: Changes must be made through a pull request.        
To https://github.com/tailcallhq/graphql-benchmarks
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'https://github.com/tailcallhq/graphql-benchmarks'
```